h5py/_errors.pyx

```diff
@@ -84,7 +84,7 @@ cdef struct err_data_t:
     H5E_error_t err
     int n
 
-cdef herr_t walk_cb(int n, H5E_error_t *desc, void *e):
+cdef herr_t walk_cb(int n, H5E_error_t *desc, void *e) with gil:
```

Comment:
I don't think this one actually does need the GIL - it's entirely manipulating a cdef-ed struct. It's also only ever called in `set_exception()`, which is called after we've reobtained the GIL. Probably it doesn't add much overhead, but for the sake of clarity, I'd prefer we only add `with gil` on functions that are likely to need it.

Reply:
There are indeed a few callback functions which could be declared "nogil". I have been considered this option to flag them as "safe for C"